src/ResourceManager/Compute/Commands.Compute/VirtualMachine/Config/SetAzureVMDataDiskCommand.cs

```diff
@@ -0,0 +1,129 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Compute.Common;
+using Microsoft.Azure.Commands.Compute.Models;
+using Microsoft.Azure.Management.Compute.Models;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Compute
+{
+    [Cmdlet(
+        VerbsCommon.Set,
+        ProfileNouns.DataDisk),
+    OutputType(
+        typeof(PSVirtualMachine))]
+    public class SetAzureVMDataDiskCommand : AzurePSCmdlet
+    {
+        private const string NameParameterSet = "ChangeWithName";
+        private const string LunParameterSet = "ChangeWithLun";
+
+        [Alias("VMProfile")]
+        [Parameter(
+            Mandatory = true,
+            Position = 0,
+            ValueFromPipeline = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = HelpMessages.VMProfile)]
+        [ValidateNotNullOrEmpty]
+        public PSVirtualMachine VM { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            Position = 1,
+            ParameterSetName = NameParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = HelpMessages.VMDataDiskName)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            Position = 3,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = HelpMessages.VMDataDiskCaching)]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet(ValidateSetValues.ReadOnly, ValidateSetValues.ReadWrite)]
+        public string Caching { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            Position = 4,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = HelpMessages.VMDataDiskSizeInGB)]
+        [AllowNull]
+        public int? DiskSizeInGB { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            Position = 5,
+            ParameterSetName = LunParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = HelpMessages.VMDataDiskLun)]
+        [ValidateNotNullOrEmpty]
+        public int? Lun { get; set; }
+
+        public override void ExecuteCmdlet()
+        {            
+            var storageProfile = this.VM.StorageProfile;            
+
+            if (storageProfile == null)
+            {
+                storageProfile = new StorageProfile();
+            }
+
+            if (storageProfile.DataDisks == null)
+            {
+                storageProfile.DataDisks = new List<DataDisk>();
```

Comment:
So, if this cmdlet fails, it would actually create an empty list of data disks in the profile.

Do you think an empty data disk list would overwrite the actual VM's data disks (if any)?